spec/abci++/abci++_properties_001_draft.md

```diff
@@ -0,0 +1,675 @@
+---
+order: 1
+title: New Methods
+---
+
+# New Methods
+
+## Description
+
+### PrepareProposal
+
+#### Parameters and Types
+
+>**TODO**: Hyperlinks for ConsensusParams, LastCommitInfo, Evidence, Event, and ValidatorUpdate are
+broken because they are defined in abci.md (and not copied over for the moment).
+
+>**TODO**: don't forget to extend the Header structure with "bool problem" when including the parts
+common with the existing ABCI spec.
+
+* **Request**:
+
+    | Name                    | Type                                        | Description                                                                                                | Field Number |
+    |-------------------------|---------------------------------------------|------------------------------------------------------------------------------------------------------------|--------------|
+    | hash                    | bytes                                       | The hash of the block to propose. This can be derived from the block header.                               | 1            |
```

Comment:
> I'm thinking that maybe it's worth including the updated hash in the response call so that Tendermint can run a sanity check on the returned values.
Do you think it's worth it?

I think it is not worth it. I don't think we should have applications calculating this piece of data if the only thing that can happen is that they get it wrong and we panic. It just doesn't seem like something that they need to really be concerned with from my perspective. If it's the same one from BeginBlock then it is the header hash. Fwiw, candidate state can just as easily be indexed by height and round, so application developers don't really require a way to make sure the two hashes match.

Reply:
Thanks for the info. I had got it wrong then. I'll update the text to clarify it's the header's hash